code/edgeidentity/src/test/java/com/adobe/marketing/mobile/edge/identity/IdentityStateTests.java

```diff
@@ -542,4 +562,296 @@ public void testUpdateLegacyExperienceCloudId_notSetWhenExistingIsNull() {
 		assertNull(state.getIdentityProperties().getECIDSecondary());
 		verify(mockSharedPreferenceEditor, Mockito.times(0)).apply();
 	}
+
+	// ======================================================================================================================
+	// Tests for method : updateAdvertisingIdentifier(final Event event, final SharedStateCallback callback)
+	// ======================================================================================================================
+
+	// With consent change
+	@Test
+	public void testupdateAdvertisingIdentifier_notSetWhenExistingIsNull() {
+		IdentityState state = new IdentityState(new IdentityProperties());
+		state.getIdentityProperties().setECID(new ECID());
+
+		state.updateLegacyExperienceCloudId(null);
+
+		assertNull(state.getIdentityProperties().getECIDSecondary());
+		verify(mockSharedPreferenceEditor, Mockito.times(0)).apply();
+	}
+
+	@Test
+	public void testUpdateAdvertisingIdentifier_whenNull_thenChangedToValid() throws Exception {
```

Comment:
Same comment on this file and EventUtilsTests and others, it seems like these changes were already merged, can you double check? You may need to merge the feature branch in this current branch for latest changes and merge any conflicts